atc/db/check_factory.go

```diff
@@ -0,0 +1,360 @@
+	if found {
+		if parentType.Version() == nil {
+			return nil, false, errors.New("parent type has no version")
+		}
+	}
+
+	timeout := c.defaultCheckTimeout
+	if to := checkable.CheckTimeout(); to != "" {
+		timeout, err = time.ParseDuration(to)
+		if err != nil {
+			return nil, false, err
+		}
+	}
+
+	variables := creds.NewVariables(
+		c.secrets,
+		checkable.TeamName(),
+		checkable.PipelineName(),
+	)
+
+	source, err := creds.NewSource(variables, checkable.Source()).Evaluate()
+	if err != nil {
+		return nil, false, err
+	}
+
+	filteredTypes := resourceTypes.Filter(checkable).Deserialize()
+	versionedResourceTypes, err := creds.NewVersionedResourceTypes(variables, filteredTypes).Evaluate()
+	if err != nil {
+		return nil, false, err
+	}
+
+	// This could have changed based on new variable interpolation so update it
+	resourceConfigScope, err := checkable.SetResourceConfig(source, versionedResourceTypes)
+	if err != nil {
+		return nil, false, err
+	}
+
+	if fromVersion == nil {
+		rcv, found, err := resourceConfigScope.LatestVersion()
+		if err != nil {
+			return nil, false, err
+		}
+
+		if found {
+			fromVersion = atc.Version(rcv.Version())
+		}
+	}
+
+	plan := atc.Plan{
+		Check: &atc.CheckPlan{
+			Name:        checkable.Name(),
+			Type:        checkable.Type(),
+			Source:      source,
+			Tags:        checkable.Tags(),
+			Timeout:     timeout.String(),
+			FromVersion: fromVersion,
+
+			VersionedResourceTypes: versionedResourceTypes,
+		},
+	}
+
+	check, created, err := c.CreateCheck(
+		resourceConfigScope.ID(),
+		resourceConfigScope.ResourceConfig().ID(),
+		resourceConfigScope.ResourceConfig().OriginBaseResourceType().ID,
+		checkable.TeamID(),
+		manuallyTriggered,
+		plan,
+	)
+	if err != nil {
+		return nil, false, err
+	}
+
+	return check, created, nil
+}
+
+func (c *checkFactory) CreateCheck(resourceConfigScopeID, resourceConfigID, baseResourceTypeID, teamID int, manuallyTriggered bool, plan atc.Plan) (Check, bool, error) {
+	tx, err := c.conn.Begin()
+	if err != nil {
+		return nil, false, err
+	}
+
+	defer Rollback(tx)
+
+	planPayload, err := json.Marshal(plan)
+	if err != nil {
+		return nil, false, err
+	}
+
+	es := c.conn.EncryptionStrategy()
+	encryptedPayload, nonce, err := es.Encrypt(planPayload)
+	if err != nil {
+		return nil, false, err
+	}
+
+	metadata, err := json.Marshal(map[string]interface{}{
+		"team_id":               teamID,
+		"resource_config_id":    resourceConfigID,
+		"base_resource_type_id": baseResourceTypeID,
+	})
+	if err != nil {
+		return nil, false, err
+	}
+
+	var id int
+	var createTime time.Time
+	err = psql.Insert("checks").
+		Columns(
+			"resource_config_scope_id",
+			"schema",
+			"status",
+			"manually_triggered",
+			"plan",
+			"nonce",
+			"metadata",
+		).
+		Values(
+			resourceConfigScopeID,
+			schema,
+			CheckStatusStarted,
+			manuallyTriggered,
+			encryptedPayload,
+			nonce,
+			metadata,
+		).
+		Suffix(`
+			ON CONFLICT DO NOTHING
```

Comment:
The conflict would only happen when it matches `CREATE UNIQUE INDEX resource_config_scope_id_key ON checks (resource_config_scope_id) WHERE status = 'started' AND manually_triggered = false;`

From an api request, the `manually_triggered` is `true` it will never error so one could create as many checks as he wants for same resource.